src/components/Explore/filters/AddFilter.test.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { AddFilter } from './AddFilter';
+import { FilterByVariable } from './FilterByVariable';
+import { reportAppInteraction, USER_EVENTS_ACTIONS, USER_EVENTS_PAGES } from '../../../utils/analytics';
+
+jest.mock('../../../utils/analytics', () => ({
+  reportAppInteraction: jest.fn(),
+  USER_EVENTS_ACTIONS: { common: { new_filter_added_manually: 'new_filter_added_manually' } },
+  USER_EVENTS_PAGES: { common: 'common' },
+}));
+
+jest.mock('./FilterRenderer', () => ({
+  FilterRenderer: jest.fn(() => <div>FilterRenderer Component</div>),
+}));
+
+const mockModel = {
+  _addWip: jest.fn(),
+  useState: jest.fn(() => ({ _wip: undefined })),
+};
+
+describe('AddFilter Component', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders add filter bar with text when no filters are present', () => {
+    render(<AddFilter model={mockModel as unknown as FilterByVariable} otherFiltersLength={0} />);
+    expect(screen.getByText('Filter by attribute...')).toBeInTheDocument();
+  });
+
+  it('renders add filter bar without text when filters are present', () => {
+    render(<AddFilter model={mockModel as unknown as FilterByVariable} otherFiltersLength={1} />);
+    expect(screen.queryByText('Filter by attribute...')).not.toBeInTheDocument();
+  });
+
+  it('renders FilterRenderer when _wip is defined', () => {
+    const mockModel = {
```

Comment:
Would it be possible to pass a real instance as the model? Something similar to:

```js
const model = new FilterByVariable({ ... })

jest.spyOn(model, 'useState').mockReturnValue({ _wip: { key: 'testKey' } });
```

Reply:
Sure, can do!